Framework/LiveData/src/ISIS/ISISHistoDataListener.cpp

```diff
@@ -124,6 +124,15 @@ bool ISISHistoDataListener::connect(const Poco::Net::SocketAddress &address) {
 
   loadTimeRegimes();
 
+  // Create dummy workspace to store instrument data
+  bufferWorkspace = WorkspaceFactory::Instance().create("Workspace2D", 1, 1, 1);
+
+  bufferWorkspace->getAxis(0)->unit() =
+      Kernel::UnitFactory::Instance().create("TOF");
+  bufferWorkspace->setYUnit("Counts");
+
+  runLoadInstrument(bufferWorkspace, getString("NAME"));
```

Comment:
This is probably causing you to work (cache) with workspaces to access the instrument. This [algorithm](https://github.com/mantidproject/mantid/blob/master/Framework/DataHandling/src/LoadInstrument.cpp) is using `InstrumentDefinitionParser` like [here](https://github.com/mantidproject/mantid/blob/master/Framework/DataHandling/src/LoadInstrument.cpp#L176). Maybe using that type directly rather than going through the algorithm would be more convenient and save lines of code.